src/Plugin/Action/ChangeOgMembershipRoleBase.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Drupal\og\Plugin\Action;
+
+use Drupal\Core\Action\ConfigurableActionBase;
+use Drupal\Core\Entity\DependencyTrait;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\user\RoleInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a base class for operations to change a user's og membership.
+ */
+abstract class ChangeOgMembershipRoleBase extends ConfigurableActionBase implements ContainerFactoryPluginInterface {
+
+  use DependencyTrait;
+
+  /**
+   * The user role entity type.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeInterface
+   */
+  protected $entityType;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeInterface $entity_type) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->entityType = $entity_type;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('entity.manager')->getDefinition('user_role')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration() {
+    return array(
+      'rid' => '',
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state) {
+    $roles = user_role_names(TRUE);
+    unset($roles[RoleInterface::AUTHENTICATED_ID]);
+    $form['rid'] = array(
+      '#type' => 'radios',
+      '#title' => t('Role'),
+      '#options' => $roles,
+      '#default_value' => $this->configuration['rid'],
+      '#required' => TRUE,
+    );
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitConfigurationForm(array &$form, FormStateInterface $form_state) {
+    $this->configuration['rid'] = $form_state->getValue('rid');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function calculateDependencies() {
+    if (!empty($this->configuration['rid'])) {
+      $prefix = $this->entityType->getConfigPrefix() . '.';
+      $this->addDependency('config', $prefix . $this->configuration['rid']);
+    }
+    return $this->dependencies;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function access($object, AccountInterface $account = NULL, $return_as_object = FALSE) {
+    return TRUE;
```

Comment:
I can imagine that this is meant to be something like "If the user '$account' has access to perform the 'add|remove' 'specific-role' to a group member'".
I do not know if there are plans to add per role and add|remove action permission (at least for now) but until then, we can check the 'administer group' permission on the account since that is needed to access the admin page currently.

Though, in the future, we should add the per role action-role permission to allow these actions to be attached to custom views outside the admin area.